src/lib/clean_number.js

```diff
@@ -16,14 +16,18 @@ var BADNUM = require('../constants/numerical').BADNUM;
 // precompile these regex's for speed
 var FRONTJUNK = /^['"%,$#\s']+/;
 var ENDJUNK = /['"%,$#\s']+$/;
+var GLOBALJUNK = /[,\s]/g;
 
 /**
  * cleanNumber: remove common leading and trailing cruft
  * Always returns either a number or BADNUM.
  */
 module.exports = function cleanNumber(v) {
     if(typeof v === 'string') {
-        v = v.replace(FRONTJUNK, '').replace(ENDJUNK, '');
+        v = v
+            .replace(FRONTJUNK, '')
+            .replace(ENDJUNK, '')
+            .replace(GLOBALJUNK, '');
```

Comment:
> The perf tests I did when playing with dates make me think it'll be faster to turn this into a single regex

Good call. I'll update this.

> Also do we really want to allow any and all internal whitespace?

Just `' '` is probably enough. It is also what we used to do here:

https://github.com/plotly/plotly.js/pull/1078/files#diff-9664b6a57e7771275ffd0c8805ea5917L30